### PR TITLE
Missing variable declarations

### PR DESCRIPTION
**Pull request by OliverFriedrich**
* Missing variable declarations
* add JSHint for global CodeMirror variable

### CODE REVIEW

File: src/pcre.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* global CodeMirror */
```
Comment (xavierog) [inline src/pcre.js:2]: Can you move this to a separate commit and add an extra comment reflecting this is intended to please JSHint?

**Review by xavierog — CHANGES_REQUESTED** [review]
Hello and thank you for your pull request. Your changes are indeed relevant. Please address the comment I left in the review and I'll approve it.

File: src/pcre.js (hunk 1 above)
Comment (OliverFriedrich) [reply src/pcre.js:2]: Sorry for asking: 
- should i delete the line, make a commit, add it again and make a new comment?
- or should i create a new branch that covers the fixes as a commit and the JSHint as a second commit?

Also i have an additional branch, that holds a dependency injection to make it compatible to Laravel/Vue.js, that is based on the current branch ...

Comment (xavierog) [reply src/pcre.js:2]: I would advise to delete the line, amend your commit, and add a second commit with:

- `/* global CodeMirror */`
- a comment in the code reflecting that line is intended for JSHint (that is not very important but that is not obvious either)
- a proper commit message

...then you can `git push --force` (Github handles that quite well).

However, if you have an additional branch of yours, you will likely need to rebase it on top of the second commit, whatever its branch.

Comment (OliverFriedrich) [reply src/pcre.js:2]: ok, done

Comment (xavierog) [reply src/pcre.js:2]: Almost! The second commit lacks this:
> a comment **in the code** reflecting that line is intended for JSHint (that is not very important but that is not obvious either)

(this is admittedly redundant with the commit message).

Comment (OliverFriedrich) [reply src/pcre.js:2]: ok, next try :)

**Comment by xavierog** [comment]
This PR is on hold and will likely be replaced with #3

**Comment by xavierog** [comment]
#3 has been accepted, closing this one too. Thanks for your contributions. I have published v1.0.6 through npm.
